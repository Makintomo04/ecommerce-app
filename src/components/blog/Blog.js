import React from "react";
import BlogData from "./BlogData";
import "./Blog.scss";
import { Link } from "react-router-dom";
const Blog = ({ blogGender }) => {
  return (
    <div className="Blog">
      <div className="Blog__heading">
        <h2>IN THE KNOW</h2>
        <p>WWW.MARQE.COM</p>
      </div>
      <div className="Blog__row">
        {!blogGender
          ? BlogData.map(({ id, imageURL, title }) => (
              <div key={id} className="Blog__card">
                <div
                  className="Blog__card__image"
                  style={{ backgroundImage: `url(${imageURL})` }}
                ></div>

                <h3 className="Blog__card__title">{title}</h3>
                <span>
                  <Link to="/">READ MORE</Link>
                </span>
              </div>
            ))
          : BlogData.filter(({ id, gender }) => gender === blogGender).map(
              ({ id, imageURL, title }) => (
                <div key={id} className="Blog__card">
                  <div
                    className="Blog__card__image"
                    style={{ backgroundImage: `url(${imageURL})` }}
                  ></div>

                  <h3 className="Blog__card__title">{title}</h3>
                  <span>
                    <Link to="/">READ MORE</Link>
                  </span>
                </div>
              )
            )}
      </div>
    </div>
  );
};

export default Blog;
