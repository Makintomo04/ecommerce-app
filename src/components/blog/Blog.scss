.Blog {
  display: flex;
  flex-direction: column;
  //   background-color: crimson;
  width: 100%;
  height: auto;
  align-items: center;
  align-content: space-between;
  padding-bottom: 80px;
  margin-bottom: 90px;
  &__heading {
    padding-bottom: 40px;
    text-align: center;
    & p {
      color: #444;
    }
  }
}

.Blog__row {
  position: relative;
  width: 100%;
  height: auto;
  display: flex;
  //   justify-content: space-between;
  //   border: 2px solid green;
}
.Blog__card {
  //   display: flex;
  //   flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 300px;
  height: 350px;
  // border: 2px solid red;
  cursor: pointer;
  padding: 10px;
  &:first-child {
    padding-left: 0;
  }
  &:last-child {
    padding-right: 0;
  }
  // background-color: rgb(222, 135, 189);
  // border: 2px solid green;
  &__image {
    min-height: 100%;
    width: 100%;
    // flex: 1;
    background-size: cover;
    background-position: center;
    margin-bottom: 10px;
    filter: saturate(0.7);
    // background-color: burlywood;
  }
  &__title {
    // flex: 0;
    // border: 2px solid pink;
    padding: 0 10px 15px;
    font-size: 18px;
    text-align: center;
    text-transform: uppercase;
    //   background-color: rgb(135, 206, 222);
  }
  & span a {
    display: flex;
    justify-content: center;
    color: #444;
    font-size: 14px;
  }
}
