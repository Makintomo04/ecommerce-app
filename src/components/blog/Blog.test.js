import React from "react"
import {shallow} from "enzyme"
import Blog from "./Blog"


it("Should render Blog component correctly", () => {
    const blog = shallow(<Blog />);
    expect(blog).toMatchSnapshot(); // check if a component renders correctly
  });