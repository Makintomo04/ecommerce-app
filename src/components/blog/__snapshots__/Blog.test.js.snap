// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Blog"
>
  <div
    className="Blog__heading"
  >
    <h2>
      IN THE KNOW
    </h2>
    <p>
      WWW.MARQE.COM
    </p>
  </div>
  <div
    className="Blog__row"
  >
    <div
      className="Blog__card"
      key="1"
    >
      <div
        className="Blog__card__image"
        style={
          Object {
            "backgroundImage": "url(apple.jpg )",
          }
        }
      />
      <h3
        className="Blog__card__title"
      >
        The new formal
      </h3>
      <span>
        <Link
          to="/"
        >
          READ MORE
        </Link>
      </span>
    </div>
    <div
      className="Blog__card"
      key="2"
    >
      <div
        className="Blog__card__image"
        style={
          Object {
            "backgroundImage": "url(field.jpg)",
          }
        }
      />
      <h3
        className="Blog__card__title"
      >
        The Helmut Depay Archive
      </h3>
      <span>
        <Link
          to="/"
        >
          READ MORE
        </Link>
      </span>
    </div>
    <div
      className="Blog__card"
      key="3"
    >
      <div
        className="Blog__card__image"
        style={
          Object {
            "backgroundImage": "url(window.jpg)",
          }
        }
      />
      <h3
        className="Blog__card__title"
      >
        Vintage
      </h3>
      <span>
        <Link
          to="/"
        >
          READ MORE
        </Link>
      </span>
    </div>
    <div
      className="Blog__card"
      key="4"
    >
      <div
        className="Blog__card__image"
        style={
          Object {
            "backgroundImage": "url(veg.jpg)",
          }
        }
      />
      <h3
        className="Blog__card__title"
      >
        Designer Spotlight: Simone Wells
      </h3>
      <span>
        <Link
          to="/"
        >
          READ MORE
        </Link>
      </span>
    </div>
  </div>
</div>
`;
